LDT/DeVeny Documentation Additions

tbowers7 opened the pull request:
Spectrograph-specific documentation for LDT/DeVeny.  This documentation was written in support of POSE grant goals for instrument support.

Includes sphinx package ``sphinx-design`` for creating "subfigures".

Documentation only; no DevSuite run.

Will add release notes once ``v1.18.0`` posts.

tbowers7 committed:
First pass at adding DeVeny-specific info

Adding a converted LaTeX -> RST block from the DeVeny User Manual
and am working my way through it to make it look good and fix all
the references and whatnot.

	modified:   doc/spectrographs/ldt_deveny.rst

tbowers7 committed:
WIP: Making my way through DeVeny documentation

Just trying to commit changes to see where I'm at as I go.

	modified:   doc/spectrographs/ldt_deveny.rst
	modified:   pypeit/spectrographs/ldt_deveny.py

tbowers7 committed:
WIP: Moving along through the DeVeny docs

	modified:   doc/spectrographs/ldt_deveny.rst
	modified:   pypeit/spectrographs/ldt_deveny.py

tbowers7 committed:
Big progress on the DeVeny docs

Some minor changes to config and package files to accomodate
subfigures.

	modified:   doc/calibrations/bias.rst
	modified:   doc/conf.py
	new file:   doc/figures/deveny_edges_DV2.png
	new file:   doc/figures/deveny_sensfunc_throughput.png
	new file:   doc/figures/deveny_spec1d.png
	new file:   doc/figures/deveny_spec1d_fluxed_A.png
	new file:   doc/figures/deveny_spec1d_fluxed_B.png
	new file:   doc/figures/deveny_spec1d_manual.png
	new file:   doc/figures/deveny_spec2d.png
	new file:   doc/figures/deveny_spec2d_manual.png
	new file:   doc/figures/deveny_tilts_2d.png
	new file:   doc/figures/deveny_tilts_spat.png
	new file:   doc/figures/deveny_tilts_spec.png
	modified:   doc/spectrographs/ldt_deveny.rst
	modified:   doc/telluric.rst
	modified:   pyproject.toml

tbowers7 committed:
WIP: Continuing cleaning the LDT/DeVeny doc

	new file:   doc/figures/deveny_findobj_qa.png
	modified:   doc/spectrographs/ldt_deveny.rst

tbowers7 committed:
Finished the DeVeny doc page

	new file:   figures/deveny_DV6_wavecal_bad.png
	new file:   figures/deveny_DV6_wavecal_good.png
	new file:   figures/deveny_flexure_qa1.png
	new file:   figures/deveny_flexure_qa2.png
	modified:   spectrographs/ldt_deveny.rst

tbowers7 committed:
Cleaned up the DeVeny docs

	modified:   doc/spectrographs/ldt_deveny.rst

tbowers7 committed:
Spelling corrections

	modified:   doc/spectrographs/ldt_deveny.rst

profxj commented on doc/spectrographs/ldt_deveny.rst:
```diff
@@ -1,18 +1,1746 @@
+    Before you get too far, it is important to understand that PypeIt
+    reorients all 2D image data (from any spectrograph) so that the spectral
+    axis is vertical with increasing wavelength corresponding to increasing
+    pixel number. In the case of DeVeny data, this amounts to a 90º CW
```
confirm this renders ok in the docs

tbowers7 replied:
<img width="660" height="349" alt="Render of this section of docs" src="https://github.com/user-attachments/assets/569ee97f-25fb-4ec8-8670-7006d7f54fa0" />

kbwestfall replied:
FWIW, you can use the `:math:` syntax to put in latex commands.

tbowers7 replied:
@kbwestfall Just to be clear, you are preferring `:math:` syntax over unicode?  I'm fine either way, just clarifying.

kbwestfall replied:
I do personally prefer using `:math:` (I find it easier to write the latex then to dig up the unicode), but I'm fine with you using either.

tbowers7 replied:
All unicode converted to ``:math:`` syntax.

profxj commented on doc/spectrographs/ldt_deveny.rst:
```diff
@@ -1,18 +1,1746 @@
+Planning Your Observations for Reduction with PypeIt
+----------------------------------------------------
+
+Because PypeIt is an automated data reduction pipeline with minimal
```
maybe refer to it as an "end-to-end" data reduction pipeline

profxj commented on doc/spectrographs/ldt_deveny.rst:
```diff
@@ -1,18 +1,1746 @@
+observing programs will already collect all of the frames necessary for
+smooth operation of the pipeline, several items bear pointing out:
+
+-  Bias frames are used to remove fixed-pattern noise in the data generate
```
"... data [and] generate..."

profxj commented on doc/spectrographs/ldt_deveny.rst:
```diff
@@ -1,18 +1,1746 @@
+
+-  Bias frames (to remove fixed-pattern noise from the data)
+
+-  Dome Flat frames (to remove pixel-to-pixel sensitivity variations)
```
and trace the slit edges?

profxj reviewed (approved):
A few minor things from me.

This docs are amazing!
And, they are likely to help folks beyond `DeVeny`.

If we point folks to them (which I will on the Slack).
And, we may wish to point other instrument docs
towards them too.  But not in this PR.

kbwestfall commented on doc/conf.py:
```diff
@@ -40,6 +40,7 @@
     'sphinx.ext.mathjax',
     'sphinx.ext.ifconfig',
     'sphinx.ext.viewcode',
+    "sphinx_subfigure",
```
Did you try using a grid with `sphinx-design`?  See here: https://sphinx-design.readthedocs.io/en/pydata-theme/grids.html .

The main reason I ask is that there are some other features in `sphinx-design` that I'd like to try using (like drop downs), and using it for both would limit our number of dependencies.  And `sphinx-design` looks like it's more actively maintained.

tbowers7 replied:
Multi-panel figures have been rendered using ``sphinx-design`` now.

kbwestfall commented on doc/spectrographs/ldt_deveny.rst:
```diff
@@ -1,18 +1,1746 @@
+.. tip::
+
+   All PypeIt command-line scripts (*e.g.*, ``pypeit_setup``) have online help
+   available using the ``-h`` option.
```
"online help available using the ``-h`` option" reads weird to me.  By "online", do you mean on the command line?

kbwestfall commented on doc/spectrographs/ldt_deveny.rst:
```diff
@@ -1,18 +1,1746 @@
+   as ``None`` in this file and are commented out. If there are
+   non-focus frames with ``frametype None`` listed, this indicates the FITS
+   keyword was not correctly set. You may either modify the affected FITS
+   headers directly using your preferred method\ [3]_ (**recommended**), or
```
Personally, I'm always wary of editing raw data, even when you know that the information is wrong.  I.e., I would actually recommend people edit the pypeit file instead (modulo the fact that this might not have worked as expected leading to your PR...).

tbowers7 replied:
I see your point.  Especially with #1971 , it makes more sense to simply edit the ``.pypeit`` file.

kbwestfall commented on doc/spectrographs/ldt_deveny.rst:
```diff
@@ -1,18 +1,1746 @@
+nights.  See :ref:`the relevant documentation <pypeit_file>` for descriptions
+of the file format and common edits a user may wish to make.  
+
+**Specific LDT/DeVeny considerations:**
```
Make this a subsection heading?  I.e., instead of making it bold, add a line of, say, '+' characters below it?

kbwestfall commented on doc/spectrographs/ldt_deveny.rst:
```diff
@@ -1,18 +1,1746 @@
+**Specific LDT/DeVeny considerations:**
+
+#. The DeVeny-specific modifications to default PypeIt reduction parameters are
+   already included in the ``LDTDeVenySpectrograph()`` class (:ref:`listed here
```
You could link to it using 

```
:class:`~pypeit.spectrographs.ldt_deveny.LDTDeVenySpectrograph`
```

tbowers7 replied:
Thank you.  I do this elsewhere... didn't occur to me.

kbwestfall commented on doc/spectrographs/ldt_deveny.rst:
```diff
@@ -1,18 +1,1746 @@
+   This command will launch a :ref:`GUI viewer <pypeit_chk_edges>` to display
+   the combined trace image along with a sobel-filtered version used to
+   identify illumination discontinuities in the spatial direction (see figure
+   below). For DeVeny data, it should identify a single, wide slit with a
```
"wide" or "long"?

tbowers7 replied:
Both!

kbwestfall commented on doc/spectrographs/ldt_deveny.rst:
```diff
@@ -1,18 +1,1746 @@
+frequently during the reduction process.
+
+
+Setup
```
I would add a quick note or link that highlights the new setup GUI somewhere in this section.

tbowers7 replied:
Ah, yes.  This part of the document was written int he pre-GUI era.

tbowers7 replied:
Doesn't look like we have much in the documentation yet about the Setup GUI.

kbwestfall replied:
Good point.  I added an issue and assigned @badpandabear .

kbwestfall commented on doc/spectrographs/ldt_deveny.rst:
```diff
@@ -1,18 +1,1746 @@
+      |  120 | SPAT0128-SLIT0120-DET01 |       128.2 |        0.535 |      3.80 |    2.121 | 146.12 |  0.065 |
+      |  120 | SPAT0231-SLIT0120-DET01 |       231.2 |        0.984 |      3.80 |    1.641 |   5.17 |  0.065 |
+
+   By default, ``pypeit_show_1dspec`` loads the first (lowest ``SPAT`` code)
```
Add a note about the new 1d spectrum viewer in ginga?  (i.e., the `--ginga` option in `pypeit_show_1dspec`)

tbowers7 replied:
Added as a ``.. tip::``.

kbwestfall commented on doc/spectrographs/ldt_deveny.rst:
```diff
@@ -1,18 +1,1746 @@
+script can be a bit cumbersome, there is a :ref:`setup script
+<pypeit_setup_coadd2d>` available that ingests the ``.pypeit`` file or reads
+FITS headers in a directory as a starting point.
+
```
I'm not sure about the description in this section:
 - I wouldn't say that coadding is more relevant to multi-slit data compared to long-slit data.
 - The primary motivation for using the 2D coadding is to combine the data in a way that accounts for spectral or spatial shifts (particularly spatial dithering of the objects along the slit) before running the object finding and extraction routines.  If the images are statistically identical (there are no shifts), the images can be combined using the `comb_id` column without needing to use the 2D coadding.

Is that consistent with your understanding?

tbowers7 replied:
Yes, thank you.

It would be helpful to add a section to the top of [the CoAdd2D doc](https://pypeit.readthedocs.io/en/develop/coadd2d.html) describing the process in the terms you used above.  I will add this as an issue (#1999).

kbwestfall replied:
Good point.  @debora-pe may be the best person to do this.

kbwestfall commented on doc/spectrographs/ldt_deveny.rst:
```diff
@@ -1,18 +1,1746 @@
+the same object. This may be because you have exposures of the same object from
+different nights or the object was placed in different locations along the slit
+in different frames, either of which precludes coadding the processed 2D
+spectral images. In this case, you may use the ``pypeit_coadd_1dspec`` script
```
Similar to a previous comment, I'm not sure this is true.  As long as you can assign wavelength and spatial coordinates to each slit (i.e., either the spectral/spatial flexure is known or negligible) I think 2D coadds work fine with both spatial and spectral shifts.

tbowers7 replied:
Fixed.  My misunderstanding.

kbwestfall commented on doc/spectrographs/ldt_deveny.rst:
```diff
@@ -1,18 +1,1746 @@
+      [[findobj]]
+         snr_thresh = 10.
+
+On the flip side, if you observed fairly bright objects want to eliminate the
```
"... objects *and* want to ..."

kbwestfall reviewed (changes requested):
These docs are excellent!  I have a few, mostly minor comments.

tbowers7 committed:
Remove unicode characters; prefer ``:math:`` syntax.

	modified:   doc/spectrographs/ldt_deveny.rst

tbowers7 committed:
Switch from sphinx-subfigure to sphinx-design

Use sphinx-design for subfigures to allow easier integration of
other fancy features from that package.

	modified:   doc/conf.py
	modified:   doc/include/dependencies_table.rst
	modified:   doc/spectrographs/ldt_deveny.rst
	modified:   pyproject.toml

tbowers7 committed:
Responding to most PR comments

Still need to think about the last few.

	new file:   doc/figures/deveny_on_ldt.png
	modified:   doc/spectrographs/ldt_deveny.rst

kbwestfall commented on doc/conf.py:
```diff
@@ -40,6 +40,7 @@
     'sphinx.ext.mathjax',
     'sphinx.ext.ifconfig',
     'sphinx.ext.viewcode',
+    "sphinx_design",
```
Thanks for making the switch!

kbwestfall commented on doc/spectrographs/ldt_deveny.rst:
```diff
@@ -1,18 +1,1799 @@
+   This command will launch a :ref:`GUI viewer <pypeit_chk_edges>` to display
+   the combined trace image along with a sobel-filtered version used to
+   identify illumination discontinuities in the spatial direction (see figure
+   below). For DeVeny data, it should identify a single, wide longslit with a
```
At the risk of harping on this, I associate "wide" and "narrow" with dimensions along the dispersion direct, and "short" and "long" with spatial extent.  So my original comment on this was mostly advocating that we try to minimize potential confusion by keeping that adjective association. FWIW, I think the sentence is more clear if you remove the word "wide."

tbowers7 replied:
Yes, I agree.  Changed.

kbwestfall reviewed (approved):
Thanks!  I'll merge this and #1997, once the latter has a 2nd review.
